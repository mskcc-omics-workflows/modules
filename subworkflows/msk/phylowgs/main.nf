include { PHYLOWGS_CREATEINPUT } from '../../../modules/msk/phylowgs/createinput/main'
include { PHYLOWGS_PARSECNVS } from '../../../modules/msk/phylowgs/parsecnvs/main'
include { PHYLOWGS_MULTIEVOLVE } from '../../../modules/msk/phylowgs/multievolve/main'
include { PHYLOWGS_WRITERESULTS } from '../../../modules/msk/phylowgs/writeresults/main'

workflow PHYLOWGS {

    take:
    ch_input_maf_and_genelevel

    main:

    ch_versions = Channel.empty()

    ch_genelevel = ch_input_maf_and_genelevel
                        .map{
                            new Tuple(it[0],it[2])
                        }

    ch_maf = ch_input_maf_and_genelevel
                        .map{
                            new Tuple(it[0],it[1])
                        }

    PHYLOWGS_PARSECNVS(ch_genelevel)

    ch_versions = ch_versions.mix(PHYLOWGS_PARSECNVS.out.versions)

    ch_maf_and_cnv = join_maf_with_cnv(ch_maf,PHYLOWGS_PARSECNVS.out.cnv)

    PHYLOWGS_CREATEINPUT(ch_maf_and_cnv)

    ch_versions = ch_versions.mix(PHYLOWGS_CREATEINPUT.out.versions)

    PHYLOWGS_MULTIEVOLVE(PHYLOWGS_CREATEINPUT.out.phylowgsinput)

    ch_versions = ch_versions.mix(PHYLOWGS_MULTIEVOLVE.out.versions)

    PHYLOWGS_WRITERESULTS(PHYLOWGS_MULTIEVOLVE.out.trees)

    ch_versions = ch_versions.mix(PHYLOWGS_WRITERESULTS.out.versions)


    emit:

    summ        = PHYLOWGS_WRITERESULTS.out.summ        // channel: [ val(meta), [ summ ] ]
    muts        = PHYLOWGS_WRITERESULTS.out.muts        // channel: [ val(meta), [ muts ] ]
    mutass      = PHYLOWGS_WRITERESULTS.out.mutass      // channel: [ val(meta), [ mutass ] ]
    versions    = ch_versions                           // channel: [ versions.yml ]
}

def join_maf_with_cnv(maf,cnv) {
        maf_channel = maf
            .map{
                new Tuple(it[0].id,it)
                }
        cnv_channel = cnv
            .map{
                new Tuple(it[0].id,it)
                }
        mergedWithKey = maf_channel
            .join(cnv_channel)
        merged = mergedWithKey
            .map{
                new Tuple(it[1][0],it[1][1],it[2][1])
            }
        return merged

}
