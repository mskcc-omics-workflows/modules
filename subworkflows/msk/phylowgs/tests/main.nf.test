nextflow_workflow {

    name "Test Subworkflow PHYLOWGS"
    script "../main.nf"
    workflow "PHYLOWGS"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows_msk"
    tag "subworkflows/phylowgs"
    tag "phylowgs"
    tag "phylowgs/parsecnvs"
    tag "phylowgs/createinput"
    tag "phylowgs/multievolve"
    tag "phylowgs/writeresults"

    config "./nextflow.config"


    test("phylowgs - gz") {

        when {
            workflow {
                """
                input[0] = Channel.value([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['unfilteredmaf'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['facets_hisens_cncf_txt'], checkIfExists: true)
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(file(workflow.out.summ[0][1]).name,
                                    file(workflow.out.muts[0][1]).name,
                                    file(workflow.out.mutass[0][1]).name).match()
                }
            )
        }
    }

    test("phylowgs - gz - stub") {

        options "-stub"

        when {
            workflow {
                """
                input[0] = Channel.value([
                    [ id:'test', single_end:false ], // meta map
                    file('unfilteredmaf'),
                    file('facets_gene_level_txt')
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(file(workflow.out.summ[0][1]).name,
                                    file(workflow.out.muts[0][1]).name,
                                    file(workflow.out.mutass[0][1]).name).match()
                }

            )
        }
    }


}
