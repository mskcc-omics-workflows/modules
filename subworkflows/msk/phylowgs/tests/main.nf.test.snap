{
    "phylowgs - gz": {
        "content": [
            "test.summ.json.gz",
            "test.muts.json.gz",
            "test.mutass.zip"
        ],
        "timestamp": "2024-03-28T19:55:03.638169"
    },
    "phylowgs - gz - stub": {
        "content": [
            "test.summ.json.gz",
            "test.muts.json.gz",
            "test.mutass.zip"
        ],
        "timestamp": "2024-03-28T19:55:13.404789"
    }
}